components/organisms/Contributors/contributors.tsx

```diff
@@ -42,14 +42,21 @@ const Contributors = ({ repositories }: ContributorProps): JSX.Element => {
   const [selectedContributors, setSelectedContributors] = useState<DbPRContributor[]>([]);
   const [selectedListIds, setSelectedListIds] = useState<string[]>([]);
   const [popoverOpen, setPopoverOpen] = useState(false);
-
+  const [isSelectAll, setIsSelectAll] = useState(false); // state to check if all contributors are selected
   const contributors = data.map((pr) => {
     return {
       host_login: pr.author_login,
       first_commit_time: pr.updated_at,
     };
   });
-
+  // check if all contributors are selected or not
+  useEffect(() => {
```

Comment:
Any reason this requires a `useEffect`? This should be driven by the check/uncheck in the `onClick` even if the paged data changes.